embassy-nrf/src/rtc.rs

```diff
@@ -210,8 +230,9 @@ impl<T: Instance> RTC<T> {
 
 impl<T: Instance> embassy::time::Clock for RTC<T> {
     fn now(&self) -> u64 {
+        // `period` MUST be read before `counter`, see comment at the top for details.
+        let period = self.period.load(Ordering::SeqCst);
         let counter = self.rtc.counter.read().bits();
```

Comment:
Hmm, I usually understand Ordering in atomic operations in relation to other operations in the same variable, which doesn't seem to be the case here. Although I'm not sure if this is wrong, I think a `compiler_fence` would make more sense here, and it would get rid of the DMB instructions. Something like

```suggestion
        let period = self.period.load(Ordering::Relaxed);
        compiler_fence(Ordering::Acquire);
        let counter = self.rtc.counter.read().bits();
```